### DISCUSSION TITLE
With web search the model searches everything I send it and not what it should actually be searching

### DISCUSSION THREAD

**Discussion by dillfrescott** [open-webui/open-webui#3155] [General]
# Bug Report

## Description

**Bug Summary:**
With web search the model searches everything I send it and not what it should actually be searching

**Steps to Reproduce:**
Open open WebUI, turn on web search and say hello. It will search "hello" or anything you give it, wasting searches.

**Expected Behavior:**
If you say "Can you find out what the weather is like in New York" It quite literally will search "Can you find out what the weather is like in New York" and not "weather in New York" like it should. This happens regardless of model used.

**Actual Behavior:**
See above.

## Environment

- **Open WebUI Version:** v0.3.4
- **Ollama (if applicable):** n/a

- **Operating System:** Windows 11
- **Browser (if applicable):** Newest Chrome

## Reproduction Details

**Confirmation:**

- [x ] I have read and followed all the instructions provided in the README.md.
- [x ] I am on the latest version of both Open WebUI and Ollama.
- [x ] I have included the browser console logs.
- [x ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
Not really applicable

**Docker Container Logs:**
Not really applicable

## Installation Method

Docker compose on Ubuntu

## Additional Information

Its using a locally running searxng in the compose stack. I followed the instructions in the docs.

**Comment by tjbck** [comment]
That's the current implementation, you'll have to explicitly toggle on and off for web searches. I agree that it would be great if the webui can intelligently decide the prompt requires web search, so PR welcome! You would also be able to implement this enh with our `Tools` feature so it would be great if you could share your code here!
